service/insolvency_service.go

```diff
@@ -305,28 +306,28 @@ func ValidateAntivirus(svc dao.Service, insolvencyResource models.InsolvencyReso
 func GenerateFilings(svc dao.Service, transactionID string) ([]models.Filing, error) {
 
 	// Retrieve details for the insolvency resource from DB
-	insolvencyResource, practitionerResourceDaos, err := svc.GetInsolvencyPractitionersResource(transactionID)
+	insolvencyResource, practitionerResources, err := svc.GetInsolvencyPractitionersResource(transactionID)
 	if err != nil {
 		message := fmt.Errorf("error getting insolvency resource from DB [%s]", err)
 		return nil, message
 	}
 
 	var filings []models.Filing
 
-	// Check for an appointed practitioner to determine if there's a 600 insolvency form
-	for _, practitioner := range practitionerResourceDaos {
-		if practitioner.Data.Appointment != nil {
-			// If a filing is a 600 add a generated filing to the array of filings
-			newFiling := generateNewFiling(insolvencyResource, nil, "600")
+	//transform practitioner resources to expected response
+	if len(practitionerResources) > 0 {
+		practitionerResourcesData := transformers.PractitionerResourceDaosToPractitionerResourceDaosResponse(practitionerResources)
+
+		// Check for an appointed practitioner to determine if there's a 600 insolvency form
```

Comment:
It looks as though the behaviour has changed slightly here - my reading is that if 2 practitioners are submitted but only one has an appointment date, the previous code would have included the data for both, whereas the new code will only include the data for the appointed one (because the check for practitioner.Data.Appointment != nil has been moved into the transformer) - can we check this / clarify the requirement please?

Reply:
This is the exact issue I was looking at with attachments. We should not have orphaned resources in the filings. The practitioners should have an associated appointment. So looks ok.

Reply:
Checked & the scenario I was wondering about (where some practitioners had appointments but not others) is not a valid one (i.e. would be rejected by ValidateInsolvencyDetails check for "Check if there is one practitioner appointed and if there is, ensure that all practitioners are appointed" when attempting to close the transaction) so this is not a change in observable behaviour for the 600 case.

However, there is another check in ValidateInsolvencyDetails "Check if attachment type is statement-of-affairs-liquidator, if true then no practitioners must be appointed, but at least one should be present" which suggests that practitioners without appointments should be included in the filings in some scenarios (& it might be simpler to include them generally and rely on the validation to have done its job so that code would only need to be changed in one place if the validation rules changed in future).

Investigating this also made me realise that following these changes the practitioner data is not being passed in to the generateNewFiling function for other filing types, so for LIQ02 & LIQ03 we are returning a nil practitioner section in the filing.  (As part of recent changes to the filings generation the requirement was clarified that practitioners data should be populated in the blocks generated for 600, LIQ02 & LIQ03 but not LRESEX.)

Reply:
I have changed the implementation to follow the exact pattern it was before now. Please let us check.Thanks

Reply:
The main change I was looking for here was to see the practitioner data passed in for the other form types, not just 600, as LIQ02 & LIQ03 need this. (My understanding is that this is true even if there are no appointments, based on the statement-of-affairs-liquidator scenario).

Reply:
Fixed.